### PR TITLE
Add Install Jetpack row in Settings

### LINKED ISSUE #5369: Jetpack CP: add a row to install Jetpack in Settings
From Settings, let's add a row "Install Jetpack" for JCP sites under a feature flag:

Screen Shot 2021-11-08 at 9 17 33 AM

The integration will be in a separate issue.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #5369 

### Description
- Add `Install Jetpack` row to app settings screen behing `jetpackConnectionPackageSupport` feature flag. The tap action will be added later when working on issue #5370 
- Add tests to ensure that `Install Jetpack` row appears only when `jetpackConnectionPackageSupport` feature flag is turned on.
- Cleanup localized strings in `SettingsViewController`.
- Reorder enum cases to match the UI of settings screen and add section titles as comments to improve readability.

### Testing instructions
- Make sure that `.jetpackConnectionPackageSupport` feature flag is turned on.
- Launch app and navigate to `Settings` screen and observe that `Install Jetpack` row is visible.
- Go through the strings in `Settings` screen and ensure that they remain unaltered.

### Screenshots

| Before | After |
| --- | --- |
|Before |  After|

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift [hunk 1]
```diff
@@ -509,20 +505,37 @@ extension SettingsViewController {
...
+        case whatsNew
         case licenses
+
+        // Other
         case deviceSettings
         case wormholy
-        case whatsNew
+
+        // Logout
+        case logout
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift:537]: Love how the enum cases are sectioned that corresponds to the settings UI!

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift [hunk 2]
```diff
@@ -571,3 +586,105 @@ extension SettingsViewController: SettingsViewPresenter {
...
+            "Log Out",
+            comment: "Log out button title"
+        )
+
+        static let hiring = NSLocalizedString(
+            "Made with love by Automattic. <a href=\"https://automattic.com/work-with-us/\">We’re hiring!</a>",
+            comment: "It reads 'Made with love by Automattic. We’re hiring!'. Place \'We’re hiring!' between `<a>` and `</a>`"
+        )
+
+        enum LogoutAlert {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift:674]: I like how the alert strings are in its own enum 👍

File: WooCommerce/WooCommerceTests/ViewRelated/Settings/SettingsViewModelTests.swift [hunk 3]
```diff
@@ -38,6 +38,36 @@ final class SettingsViewModelTests: XCTestCase {
         XCTAssertTrue(viewModel.sections.count > 0)
     }
 
+    func test_sections_contain_install_jetpack_row_when_JCP_support_feature_flag_is_on() {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewRelated/Settings/SettingsViewModelTests.swift:41]: nice tests!

**Review by jaclync — APPROVED** [review]
Thanks a lot for contributing @selanthiraiyan! The code looks great with unit tests 😍 Code & testing 💯 

Just a minor suggestion that maybe the localization refactoring part could be in a separate PR. But the diffs are pretty small so all good here!

**Comment by selanthiraiyan** [comment]
Thank you for the detailed review @jaclync ❤️ 

> Just a minor suggestion that maybe the localization refactoring part could be in a separate PR. But the diffs are pretty small so all good here!

Makes complete sense. I will make sure to follow this in future.
I can also work on creating a separate PR for these changes if you want me to do that. (I will need a couple of days though)
Kindly let me know. Thanks again!

**Comment by jaclync** [comment]
> Makes complete sense. I will make sure to follow this in future.
I can also work on creating a separate PR for these changes if you want me to do that. (I will need a couple of days though)
Kindly let me know. Thanks again!

No problem at all! Merging the PR now 🚀 thank you again for your contribution 😃

**Pull request merged**